### ISSUE TITLE
Only first five tests compile

### ISSUE THREAD

**Issue by mostlyharmlessone** [pid1/test-sec-flags#12]
I ran the following:
git clone https://github.com/pid1/test-sec-flags.git --recursive
cd test-sec-flags
make setup
./test-xz
(which seemed to run fine, but I interrupted it so I could just run everything with:)
make

The last command compiled and ran the first 5 tests of unixbench, but then terminated with:

`make[2]:``Leaving directory '/home/mostlyh/test-sec-flags/unixbench/UnixBench'
make[1]: Leaving directory '/home/mostlyh/test-sec-flags/unixbench/UnixBench'
Compilation finished, running test 6
make: *** [Makefile:10: test-unixbench] Error 1`

running on Arch, updated a few days ago, i7-4930K

**Comment by xf0e** [comment]
got the same problem.

**Comment by pid1** [comment]
@mostlyharmlessone Please pull down a fresh copy of the repository and re-run make. Also, please include the relevant Makefiles.

@xf0e Please follow the steps above, and include your hardware and exact steps to reproduce the behavior.

**Comment by xf0e** [comment]
git clone https://github.com/pid1/test-sec-flags.git --recursive

```
cd test-sec-flags
make setup
./test-ffmpeg 
```

and then:

`LD      libavcodec/libavcodec.so.57
libavcodec/libopenjpegdec.o: In function`libopenjpeg_decode_frame':
libopenjpegdec.c:(.text+0x1c9): undefined reference to `opj_create_decompress'
libopenjpegdec.c:(.text+0x1e7): undefined reference to`opj_set_error_handler'
libopenjpegdec.c:(.text+0x201): undefined reference to `opj_set_warning_handler'
libopenjpegdec.c:(.text+0x21b): undefined reference to`opj_set_info_handler'
libopenjpegdec.c:(.text+0x248): undefined reference to `opj_setup_decoder'
libopenjpegdec.c:(.text+0x252): undefined reference to`opj_stream_default_create'
libopenjpegdec.c:(.text+0x272): undefined reference to `opj_stream_set_read_function'
libopenjpegdec.c:(.text+0x281): undefined reference to`opj_stream_set_skip_function'
libopenjpegdec.c:(.text+0x290): undefined reference to `opj_stream_set_seek_function'
libopenjpegdec.c:(.text+0x2a2): undefined reference to`opj_stream_set_user_data'
libopenjpegdec.c:(.text+0x2ae): undefined reference to `opj_stream_set_user_data_length'
libopenjpegdec.c:(.text+0x2be): undefined reference to`opj_read_header'
libopenjpegdec.c:(.text+0x3e8): undefined reference to `opj_create_decompress'
libopenjpegdec.c:(.text+0x3fe): undefined reference to`opj_image_destroy'
libopenjpegdec.c:(.text+0x408): undefined reference to `opj_stream_destroy'
libopenjpegdec.c:(.text+0x410): undefined reference to`opj_destroy_codec'
libopenjpegdec.c:(.text+0x621): undefined reference to `opj_decode'
libavcodec/libopenjpegdec.o: In function`libopenjpeg_decode_init':
libopenjpegdec.c:(.text.unlikely+0xd): undefined reference to `opj_set_default_decoder_parameters'
libavcodec/libopenjpegdec.o: In function`libopenjpeg_static_init':
libopenjpegdec.c:(.text.unlikely+0x2e): undefined reference to `opj_version'
libavcodec/libopenjpegenc.o: In function`libopenjpeg_encode_frame':
libopenjpegenc.c:(.text+0xe6f): undefined reference to `opj_create_compress'
libopenjpegenc.c:(.text+0xe8d): undefined reference to`opj_set_error_handler'
libopenjpegenc.c:(.text+0xea3): undefined reference to `opj_set_warning_handler'
libopenjpegenc.c:(.text+0xed4): undefined reference to`opj_stream_destroy'
libopenjpegenc.c:(.text+0xedc): undefined reference to `opj_destroy_codec'
libopenjpegenc.c:(.text+0x134c): undefined reference to`opj_set_info_handler'
libopenjpegenc.c:(.text+0x1365): undefined reference to `opj_setup_encoder'
libopenjpegenc.c:(.text+0x13a5): undefined reference to`opj_stream_default_create'
libopenjpegenc.c:(.text+0x13d1): undefined reference to `opj_stream_set_write_function'
libopenjpegenc.c:(.text+0x13e0): undefined reference to`opj_stream_set_skip_function'
libopenjpegenc.c:(.text+0x13ef): undefined reference to `opj_stream_set_seek_function'
libopenjpegenc.c:(.text+0x13fe): undefined reference to`opj_stream_set_user_data'
libopenjpegenc.c:(.text+0x140d): undefined reference to `opj_start_compress'
libopenjpegenc.c:(.text+0x141c): undefined reference to`opj_encode'
libopenjpegenc.c:(.text+0x146b): undefined reference to `opj_end_compress'
libavcodec/libopenjpegenc.o: In function`libopenjpeg_encode_close':
libopenjpegenc.c:(.text.unlikely+0xa): undefined reference to `opj_image_destroy'
libavcodec/libopenjpegenc.o: In function`libopenjpeg_encode_init':
libopenjpegenc.c:(.text.unlikely+0x44): undefined reference to `opj_set_default_encoder_parameters'
libopenjpegenc.c:(.text.unlikely+0x10f): undefined reference to`opj_image_destroy'
libopenjpegenc.c:(.text.unlikely+0x4f3): undefined reference to `opj_image_create'
/usr/bin/ld: libavcodec/libavcodec.so.57: hidden symbol`opj_read_header' isn't defined
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make: **\* [library.mak:112: libavcodec/libavcodec.so.57] Error 1`

**Comment by mostlyharmlessone** [comment]
Same result here. with make setup ; make

**Comment by anthraxx** [comment]
sure you have all ffmpeg depends and makedepends installed? f.e. `openjpeg`

**Comment by mostlyharmlessone** [comment]
Yes, when I get back to it.  However, if it's a dependency, perhaps it should be stated explicitly.

**Comment by anthraxx** [comment]
@mostlyharmlessone it is documented: https://github.com/pid1/test-sec-flags/blame/master/README.rst#L21

maybe the listed packages should be put there instead of a link... will do later when i have time :santa:

**Comment by pid1** [comment]
Updated in 635b754.

**Comment by smirkybg** [comment]
https://trac.ffmpeg.org/attachment/ticket/5694/ffmpeg_opj2.patch
This is the patch for ffmpeg.
I downloaded it and added
patch -p1 -i ffmpeg_opj2.patch
inside the test-ffmpeg script, right before starting the tests (below the curl).
I also changed the ffmpeg version to be 3.1.1, since the patch is for the latest version.
However, it still fails at the end of the second test

LD  libavutil/libavutil.so.55
collect2: error: ld returned 1 exit status
make: **\* [library.mak:100: libavutil/libavutil.so.55] Error 1
make: **\* Waiting for unfinished jobs....
STRIP   libavcodec/x86/vp9itxfm.o
skipping strip -wN ..@\* libavcodec/x86/vp9itxfm.o

Bummer...

**Comment by mostlyharmlessone** [comment]
Made sure all the dependencies were installed, tried again:
make setup
./test-ffmpeg
....
.....
libavcodec/libopenjpegenc.o: In function `libopenjpeg_encode_init':
libopenjpegenc.c:(.text.unlikely+0x44): undefined reference to`opj_set_default_encoder_parameters'
libopenjpegenc.c:(.text.unlikely+0x10f): undefined reference to `opj_image_destroy'
libopenjpegenc.c:(.text.unlikely+0x4f3): undefined reference to`opj_image_create'
/usr/bin/ld: libavcodec/libavcodec.so.57: hidden symbol `opj_read_header' isn't defined
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make: **\* [library.mak:112: libavcodec/libavcodec.so.57] Error 1
make: **\* Waiting for unfinished jobs..

Hmm

**Comment by kido5217** [comment]
I had the same error, but then I remembered that README tells us to use a clean chroot.
So I did just that and ffmpeg now compiles without a problem.
To do so I've used tools from [arch-install-scripts](https://www.archlinux.org/packages/extra/any/arch-install-scripts/)

1). make clean chroot

```
$ mkdir -p /home/kido/temp/chroot
$ sudo pacstrap -i -d /home/kido/temp/chroot/ base base-devel
$ sudo arch-chroot /home/kido/temp/chroot/ /bin/bash
```

2). configure chroot

```
# pacman -S git python python-matplotlib python-numpy
# pacman -S ffmpeg alsa-lib dcadec fontconfig fribidi gsm lame libass libavc1394 libbluray libiec61883 libmodplug libpulse libsoxr libssh libtheora libva libvdpau libwebp netcdf opencore-amr openjpeg opus schroedinger sdl speex v4l-utils xvidcore libvidstab.so libvorbis.so libvorbisenc.so libvpx.so libx264.so libx265.so hardening-wrapper ladspa libvdpau yasm
# useradd -m -g users -s /bin/bash testuser
# su - testuser
```

3). follow README

**Comment by hobindar** [comment]
I'm having the same issue as @smirkybg (same output, didn't need patch). Tried with a clean chroot, no dice. ffmpeg consistently fails on the third test.

**Comment by pid1** [comment]
@smirkybg Caught this. There is an issue upstream. https://trac.ffmpeg.org/ticket/5694

I'm just going to change the ffmpeg test to pull either 3.1.1 or git HEAD.

**Comment by pid1** [comment]
There have been multiple test suites mentioned in this bug report. It looks like we have a fix for the ffmpeg issues, but please keep all future comments related to the Unixbench issue that was first reported by @mostlyharmlessone. If you are having trouble with any of the other test suites, open a new issue.

**Comment by pid1** [comment]
Closing as deprecated, as we are moving ahead with the flags.
